test/e2e/util.go

```diff
@@ -787,10 +787,14 @@ func testContainerOutputInNamespace(scenarioName string, c *client.Client, pod *
 		Failf("Failed to create pod: %v", err)
 	}
 
-	containerName := pod.Spec.Containers[0].Name
-
 	// Wait for client pod to complete.
-	expectNoError(waitForPodSuccessInNamespace(c, pod.Name, containerName, ns))
+	var containerName string
+	for id, container := range pod.Spec.Containers {
+		expectNoError(waitForPodSuccessInNamespace(c, pod.Name, container.Name, ns))
+		if id == containerIndex {
+			containerName = container.Name
+		}
+	}
```

Comment:
I still think you need to check that containerName is not empty here, to fail immediately and clearly if a test passes an invalid containerIndex.